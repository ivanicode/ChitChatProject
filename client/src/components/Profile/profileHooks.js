import { useContext, useEffect, useState } from 'react';
import { MyContext } from '../Home/Home'
import {hobby} from '../RegisterProfile/dictionary';

export function useProfileHooks(){
    const { partnerData } = useContext(MyContext)

    const [age, setAge] = useState();
    const [hobbys, setHobbys] = useState([])

    useEffect(() => {
        if(partnerData?.birth_date){
            const now = new Date();
            const birthDate = new Date(partnerData.birth_date)
            setAge(now.getFullYear() - birthDate.getFullYear())
        }
        if(partnerData?.interests){
            const chosenHobbys = partnerData.interests;
            setHobbys(hobby.filter((element) => chosenHobbys.includes(element.id)).map(el => el?.label))
        }
    }, [partnerData])

    return {age, partnerData, hobbys}
}