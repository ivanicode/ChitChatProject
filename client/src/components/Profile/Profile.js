import React from 'react';
import { useProfileHooks } from './profileHooks'


export function Profile () {
    
    const { age, partnerData, hobbys } = useProfileHooks()

    return (
        <div className="partnerProfile">
            <div className="partnerPicture"></div>
            <div className="partnerNicknameAndStatusDiv">
                <div className="partnerNickname">
                    {partnerData.nickname}
                </div>
                <div className="partnerStatus">
                    <img src={'/' + partnerData?.relationship + '.png'}></img>
                </div>
            </div>
            <div className="partnerInformations">               
                <div className="partnerCity">{partnerData.city}</div>
                <div className="partnerAge">{age}</div>
                <div className="partnerInterests">{hobbys.join(', ')}</div>
            </div>
        </div>
    );   
}

export default Profile;
