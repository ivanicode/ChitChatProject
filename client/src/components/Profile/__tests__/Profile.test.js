import React from 'react';
import {shallow} from 'enzyme';
import {Profile} from '../Profile';


describe('Profile component', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<Profile />)
        expect(wrapper).toMatchSnapshot()
    })
})
