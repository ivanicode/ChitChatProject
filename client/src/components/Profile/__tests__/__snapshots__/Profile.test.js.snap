// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should match snapshot 1`] = `
<div
  className="profile"
>
  <div
    className="mainPictureOfHim"
  />
  <div
    className="statusNicknameOfHimDiv"
  >
    <div
      className="nicknameOfHim"
    >
      Nickname
    </div>
    <div
      className="statusOfHim"
    />
  </div>
  <div
    className="interestsAndTheRestOfHim"
  >
    <div
      className="cityOfHim"
    >
      Miasto
    </div>
    <div
      className="ageOfHim"
    >
      Wiek
    </div>
    <div
      className="interestsOfHim"
    >
      Główne zainteresowania po przecinku(max 3)
    </div>
  </div>
  <div
    className="myPageOfHim"
  />
</div>
`;
